### PR TITLE
Multi decade fixes

### PR DESCRIPTION
**Pull request by lisazeyen**
Smaller fixes to multi-decade optimisation

## Changes proposed in this Pull Request

- rescale buses marginal prices also by investment period weightings
- add documenation for multi-decade investment
- add investment period and investment period weightings to documentation
- add attribute carrier to component "`Load`"
- rename column of `investment_period_weightings` from "`time`" -> "`years`"
- make type of attribute "`investment_period`" in global constraint a float, default value is `NaN` and results in applying the constraint to all investment periods
- fix bug if global constraint is only valid for a given investment period

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Unit tests for new features were added (if applicable).
- [x] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

File: pypsa/component_attrs/networks.csv [hunk 1]
```diff
@@ -1,7 +1,9 @@
...
+investment_periods,pandas.Index,years,[],"Time periods of investment. Only used for multi investment optimisation. Years have to be integer and increasing (e.g. [2025,2030,2035]). Default is an empty pd.Index([]).",Input (optional)
+investment_period_weightings,pandas.DataFrame,n/a,1,Weightings applied to each investment period. Objective weightings are multiplied with all cost coefficients in the objective function of the respective investment period (e.g. used to include social discount rate). Years weighting denote the elapsed time until the subsequent investment period (e.g. used for global constraints CO2 emissions).,Input (optional)
 now,any,n/a,"""now""","The current snapshot/time/scenario, relevant e.g. when ``network.pf()`` is called without a snapshot argument.",Input (optional)
```
Comment (FabianHofmann) [inline pypsa/component_attrs/networks.csv:7]: Good catch

File: pypsa/linopf.py [hunk 2]
```diff
@@ -646,14 +646,14 @@ def define_global_constraints(n, sns):
...
-        period_weighting = n.investment_period_weightings["time"]
+        period_weighting = n.investment_period_weightings["years"]
         weightings = n.snapshot_weightings.mul(period_weighting, level=0, axis=0).loc[sns]
     else:
         weightings = n.snapshot_weightings.loc[sns]
 
     def get_period(n, glc, sns):
         period = slice(None)
-        if n._multi_invest and (glc["investment_period"] != ''):
+        if n._multi_invest and (~np.isnan(glc["investment_period"])):
```
Comment (FabianHofmann) [inline pypsa/linopf.py:656]:
```suggestion
        if n._multi_invest and not np.isnan(glc["investment_period"]):
```

**Review by FabianHofmann — APPROVED** [review]
@lisa, I had a look through all changes and everything seems to be fine :)

File: doc/release_notes.rst [hunk 3]
```diff
@@ -13,43 +13,43 @@ Upcoming Release
...
+* ``PyPSA`` now supports the optimization of multiple investment periods, also known as pathway optimization. With this feature, snapshots can span over multiple years or decades which are divided into investment periods. Within each investment period, assets can be added to the network. The optimization only works with ``pyomo=False``. For more information see the `example notebook <https://pypsa.readthedocs.io/en/latest/examples/multi-investment-optimisation.html>`_.
 
-* A new attribute (column) "carrier" was added to the components ``Line``, ``Link`` and ``Store``, defining the energy carrier of the components. It's default is an empty string. When calling ``network.calculate_dependant_values``, empty carriers are replaced by the carriers of the buses to which the components are attached. 
+* A new attribute (column) "carrier" was added to the components ``Line``, ``Link``,``Store`` and ``Load``, defining the energy carrier of the components. It's default is an empty string. When calling ``network.calculate_dependant_values``, empty carriers are replaced by the carriers of the buses to which the components are attached.
```
Comment (fneum) [inline doc/release_notes.rst:52]:
```suggestion
* A new attribute (column) "carrier" was added to the components ``Line``, ``Link``, ``Store`` and ``Load``, defining the energy carrier of the components. Its default is an empty string. When calling ``network.calculate_dependent_values()``, empty carriers are replaced by the carriers of the buses to which the components are attached. 
```

**Review by fneum — APPROVED** [review]
Also just one mini comment for the release notes! Good to go!

**Comment by lisa** [comment]
> @lisa, I had a look through all changes and everything seems to be fine :) Just a few suggestions

# `@lisa` User Notification

Hello! You're receiving this notice because you've tagged me, @lisa, on GitHub, likely in error. Lisa is a common first name, and happens to be mine. I use the @lisa username here on GitHub. While being tagged in various repositories is an interesting way for me to discover new projects, it's likely not your intention. Did you mean to tag someone else? Did you mean to use `@lisa` instead, in a [word as word](https://en.wikipedia.org/wiki/Use%E2%80%93mention_distinction) usage?

# Next Steps

I'm unsubscribing from this issue, pull request or project so if you meant to tag someone else to notify them, I would recommend that. If you meant to use my user ID as a plain string, without having GitHub notify me, you can do so by wrapping with backticks (the \` character), like so: `@lisa`.

[this text](https://gist.github.com/lisa/2b36dad1c3eaba7c72a5a57cfb7a50c3)

**Pull request merged**